src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEngine.Unix.cs

```diff
@@ -12,50 +12,13 @@ namespace System.Net.Sockets
 {
     internal sealed unsafe class SocketAsyncEngine : IThreadPoolWorkItem
     {
-        //
-        // Encapsulates a particular SocketAsyncContext object's access to a SocketAsyncEngine.
-        //
-        public readonly struct Token
-        {
-            private readonly SocketAsyncEngine? _engine;
-            private readonly IntPtr _handle;
-
-            public Token(SocketAsyncContext context)
-            {
-                AllocateToken(context, out _engine, out _handle);
-            }
-
-            public bool WasAllocated
-            {
-                get { return _engine != null; }
-            }
-
-            public void Free()
-            {
-                if (WasAllocated)
-                {
-                    _engine!.FreeHandle(_handle);
-                }
-            }
-
-            public bool TryRegister(SafeSocketHandle socket, out Interop.Error error)
-            {
-                Debug.Assert(WasAllocated, "Expected WasAllocated to be true");
-                return _engine!.TryRegister(socket, _handle, out error);
-            }
-        }
-
         private const int EventBufferCount =
 #if DEBUG
             32;
 #else
             1024;
 #endif
 
-        private static readonly object s_lock = new object();
```

Comment:
removing this global lock is going to be beneficial for any scenarios where there is a lot of short-living socket connections

@davidfowl @halter73 perhaps this would be useful for reverse proxy?

Reply:
> for any scenarios where there is a lot of short-living socket connections

Often connections are long-lived because security handshake is time consuming.